tests/phpunit/integration/Core/Authentication/Has_Multiple_AdminsTest.php

```diff
@@ -46,7 +47,64 @@ public function test_get() {
 		$this->factory()->user->create( array( 'role' => 'administrator' ) );
 
 		// Delete the transient and check that we have multiple admins now.
-		$this->transients->delete( Has_Multiple_Admins::OPTION );
 		$this->assertTrue( $has_multiple_admins->get(), 'Should return true when there are multiple admin users' );
 	}
+
+	public function test_register__user_register_hook() {
+		$has_multiple_admins = new Has_Multiple_Admins( $this->transients );
+		$has_multiple_admins->register();
+
+		// Initially, there is only one admin (ID=1).
+		$this->assertFalse( $has_multiple_admins->get(), 'Should return false when there is only one admin user' );
+
+		// Create a new user with non-admin role, should not affect the transient.
+		$this->factory()->user->create( array( 'role' => 'editor' ) );
+		$this->assertFalse( $has_multiple_admins->get(), 'Should still return false when a non-admin user is created' );
+
+		// Create a new admin user, should clear the transient and reflect the change.
+		$this->factory()->user->create( array( 'role' => 'administrator' ) );
+		$this->assertTrue( $has_multiple_admins->get(), 'Should return true when a second admin user is created' );
+	}
+
+	public function test_register__deleted_user_hook() {
+		$has_multiple_admins = new Has_Multiple_Admins( $this->transients );
+		$has_multiple_admins->register();
+
+		// Create a second admin user and one non-admin user.
+		$admin_id_2   = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		$non_admin_id = $this->factory()->user->create( array( 'role' => 'editor' ) );
+
+		// Now we have multiple admins.
+		$this->assertTrue( $has_multiple_admins->get(), 'Should return true when there are multiple admin users' );
+
+		// Delete a non-admin user, should not affect the transient.
+		wp_delete_user( $non_admin_id );
+		$this->assertTrue( $has_multiple_admins->get(), 'Should still return true when a non-admin user is deleted' );
+
+		// Delete one of the admin users, should clear the transient and reflect the change.
+		wp_delete_user( $admin_id_2 );
+		$this->assertFalse( $has_multiple_admins->get(), 'Should return false when there is only one admin user left' );
+	}
+
+	public function test_register__set_user_role_hook() {
```

Comment:
As noted before, we'll also want to test that adding and removing a role has the same effect. The test for `set_user_role` will show that the extra hook shouldn't be necessary.